cmd/osm-controller/osm-controller.go

```diff
@@ -228,7 +228,7 @@ func main() {
 	httpServer := httpserver.NewHTTPServer(constants.OSMHTTPServerPort)
 
 	// Health/Liveness probes
-	funcProbes := []health.Probes{xdsServer}
+	funcProbes := []health.Probes{xdsServer, smi.HealthChecker{SMIKubeConfig: kubeConfig}}
```

Comment:
I would suggest constructing the client for the apiextensions API here from the config, then initializing the SMI health checker with that instead of the REST config. Then you have the option to construct the health checker with the [faked client](https://pkg.go.dev/k8s.io/apiextensions-apiserver@v0.21.0/pkg/client/clientset/clientset/fake#NewSimpleClientset) which would allow you to practically unit test the new probe.